sql/analyzer/resolve_tables.go

```diff
@@ -81,38 +81,45 @@ func resolveTables(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql
 	})
 }
 
-func resolveTable(ctx *sql.Context, t *plan.UnresolvedTable, a *Analyzer) (sql.Node, error) {
+func resolveTable(ctx *sql.Context, t sql.VersionedTable, a *Analyzer) (sql.Node, error) {
```

Comment:
This doesn't seem right, we have no idea if this is a versioned table or not when we run this rule

Reply:
Having read the rest, I think I see what's going on.

Would just rename this to sql.UnresolvedTable instead